File: ts/src/Evals/02-use-braintrust-objects/use-braintrust-objects.ts
import { loadPrompt, Eval, initFunction, initDataset } from "braintrust";
import {ExactMatch, NumericDiff} from "autoevals"
import dotenv from "dotenv";
import * as path from "path";

// Load the single .env at the repo root (02-use-braintrust-objects -> Evals -> src -> ts -> root)
dotenv.config({ path: path.resolve(__dirname, "../../../../.env") });

const PROJECT_NAME: string = process.env.BRAINTRUST_PROJECT || "multiturn-agent";

Eval(
    PROJECT_NAME, 
    {
        data: initDataset(PROJECT_NAME, {dataset: "Countries"}),
        task: initFunction({projectName: PROJECT_NAME, slug: "country-structured-prompt"}),
        scores: [
            ExactMatch,
            NumericDiff
        ]
    }
)

// export BRAINTRUST_API_KEY=<YOUR_API_KEY>
// npx braintrust eval src/Evals/02-use-braintrust-objects/use-braintrust-objects.ts

File: py/src/evals/02_use_braintrust_objects/use_braintrust_objects.py
import os
from pathlib import Path
from braintrust import Eval, init_function, init_dataset
from autoevals import ExactMatch, NumericDiff
from dotenv import load_dotenv

repo_root = Path(__file__).resolve().parents[4]  # file -> 02_use_braintrust_objects -> evals -> src -> py -> root
load_dotenv(repo_root / ".env")

PROJECT_NAME = os.getenv("BRAINTRUST_PROJECT")

Eval(
    PROJECT_NAME,
    data=init_dataset(PROJECT_NAME, name="Countries"),
    task=init_function(project_name=PROJECT_NAME, slug="country-structured-prompt"),
    scores=[
        ExactMatch,
        NumericDiff,
    ],
)
